### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by laurentmahieu**

### CODE REVIEW

**Review by BlandineM — DISMISSED** [review]

File: backend/route/auth.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          details: "Either mail or password is incorrect",
+          message: infoAuth
+        });
+
+      const token = jwt.sign(user, jwtSecret);
+      return res.status(200).json({ user, token });
+    }
+  )(req, res);
+});
+router.get(
```
Comment (LoicBrassart) [inline backend/route/auth.js]: No... This one doesn't serve any purpose in production, please delete !

File: frontend/src/pages/Login.jsx [hunk 2]
```diff
@@ -1,14 +1,44 @@
...
+  const sublogin = e => {
+    e.preventDefault();
+    axios
+      .post("http://localhost:5000/auth/login", {
+        mail: email,
+        password
+      })
+      .then(({ data }) => {
+        console.log(data);
+        //TODO redirect to "/"
```
Comment (LoicBrassart) [inline frontend/src/pages/Login.jsx]: Do not add TODOs: either you do it, or you add it to the backlog, but that's not the place to note this

File: frontend/src/reducer/reducer.js [hunk 3]
```diff
@@ -36,6 +36,14 @@ const reducer = (state = initialState, action) => {
         ...newState
       };
 
+    case "FETCHING_USER_DATA":
```
Comment (LoicBrassart) [inline frontend/src/reducer/reducer.js]: Whet does this do ? I'm honestly confused

File: frontend/src/store/store.js [hunk 4]
```diff
@@ -76,6 +76,11 @@ const initialState = {
         km: "26 450 km"
       }
     ]
+  },
+  user: {
```
Comment (LoicBrassart) [inline frontend/src/store/store.js]: Now is not really the best time to add false values to your project, is it ?

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Some things to check out before I accept this

**Review by BlandineM — DISMISSED** [review]

File: backend/route/auth.js (hunk 1 above)
Comment (laurentmahieu) [reply backend/route/auth.js]: but no server without I don't know why

File: frontend/src/reducer/reducer.js (hunk 3 above)
Comment (laurentmahieu) [reply frontend/src/reducer/reducer.js]: that replace the last value by the correct user value

File: frontend/public/index.html [hunk 5]
```diff
@@ -24,7 +24,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>MemoraCar</title>
+    <title>shinyHedgehog</title>
```
Comment (LoicBrassart) [inline frontend/public/index.html]: It'd have been better to use the env key containing the name, not replace on hard-coded value with another

File: frontend/src/pages/Login.jsx [hunk 6]
```diff
@@ -18,17 +18,10 @@ function Login(props) {
...
         history.push("/");
-        //TODO store user in redux
-        //TODO store token in redux or cookies
         props.dispatch({ type: "FETCHING_USER_DATA", value: { data } });
-        console.log(props.user);
       })
-      .catch(() => {
-        //TODO notif fail
-      });
+      .catch(() => {});
```
Comment (LoicBrassart) [inline frontend/src/pages/Login.jsx]: If you don't do anything in the `catch()`, remove it entirely

File: frontend/src/pages/Signup.jsx [hunk 7]
```diff
@@ -3,18 +3,26 @@ import "./style/LoginSignup.scss";
...
   const [confirmPassword, setConfirmPassword] = useState("");
 
   function signup(e) {
     e.preventDefault();
-    axios.post("http://localhost:5000/auth/signup", {
-      mail: email,
-      password
-    });
+    axios
+      .post("http://localhost:5000/auth/signup", {
```
Comment (LoicBrassart) [inline frontend/src/pages/Signup.jsx]: Variabalize this

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Some minor changes

**Review by BlandineM — DISMISSED** [review]

**Review by LoicBrassart — DISMISSED** [review]
SnE

**Review by BlandineM — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
SnE

**Pull request merged**